core/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryParser.java

```diff
@@ -130,8 +130,13 @@ public GeoDistanceQueryBuilder fromXContent(QueryParseContext parseContext) thro
                 } else if ("validation_method".equals(currentFieldName)) {
                     validationMethod = GeoValidationMethod.fromString(parser.text());
                 } else {
-                    point.resetFromString(parser.text());
-                    fieldName = currentFieldName;
+                    if (fieldName == null) {
+                        point.resetFromString(parser.text());
+                        fieldName = currentFieldName;
+                    } else {
+                        throw new ParsingException(parser.getTokenLocation(), "[" + GeoDistanceQueryBuilder.NAME + "] point specified twice. [" + currentFieldName
```

Comment:
I'm good with the added parsing validation. Minor question, do we want to change the exception to something more generic? Something like, " malformed point ". That way we're not assuming a point was specified twice when it could have been some other autogenerated garbage?